main.go

```diff
@@ -295,6 +302,14 @@ func (o *Options) Validate() error {
 		return fmt.Errorf("--db-only-mode cannot be set to false (deprecated flag soon to be removed, feature now mandatory)")
 	}
 
+	if o.CRTimeRoundingFactor > maxCRTimeRoundingFactor {
+		return fmt.Errorf("please provide a value smaller than %v for --component-readiness-time-rounding-factor", maxCRTimeRoundingFactor)
+	}
+	if o.CRTimeRoundingFactor > 0 {
+		api.CacheRoundingDuration = o.CRTimeRoundingFactor
+		metrics.CRTimeRoundingFactor = o.CRTimeRoundingFactor
+	}
```

Comment:
Ah this is the part [I was missing](https://github.com/openshift/sippy/pull/1452/files#r1468042657), the main is modifying the values in the other packages.  Personally I'm not a big fan of global state like this

Reply:
There's a CacheOption struct in the BQ Client, or it could just be passed into the functions.  Does that make it too complicated?